mathesar_ui/src/component-library/toast/ToastController.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  faCheck,
+  faExclamationTriangle,
+  faSpinner,
+} from '@fortawesome/free-solid-svg-icons';
+import type { IconDefinition } from '@fortawesome/free-solid-svg-icons';
+import type { SvelteComponent } from 'svelte';
+import { linear } from 'svelte/easing';
+import type { Writable, Readable } from 'svelte/store';
+import { writable, derived } from 'svelte/store';
+import type { PauseableTweened } from '../common/utils/pauseableTweened';
+import { pauseableTweened } from '../common/utils/pauseableTweened';
+import type { IconFlip, IconRotate } from '../icon/Icon.d';
+
+/**
+ * Allows control of the toast message after it is displayed
+ */
+interface ToastEntryController {
+  id: number,
+  progress: PauseableTweened,
+  dismiss: () => void,
+}
+
+interface Icon {
+  data: IconDefinition,
+  spin?: boolean,
+  flip?: IconFlip,
+  rotate?: IconRotate,
+}
+
+interface ToastEntryProps {
+  /**
+   * When given, the new toast message will replace an existing toast message
+   * with the specified id.
+   */
+  id?: number,
+  title?: Readable<string> | string,
+  message?: Readable<string> | string,
+  /**
+   * If provided, will be used in place of `title` and `message`.
+   */
+  contentComponent?: typeof SvelteComponent,
+  contentComponentProps?: Readable<unknown> | unknown,
+  icon?: Readable<Icon> | Icon,
+  backgroundColor: Readable<string> | string,
+  textColor: Readable<string> | string,
+  progressColor: Readable<string> | string,
```

Comment:
Can we remove these and replace them with a prop that sets css classes on the wrapper element?

I am not a fan of programmatically styling elements unless warranted. I would also like to avoid inline styling as much as possible. In this case, using css classes would enable us to customize css separately.

It would be essential for us to be able to create a separate css library, for theming and for reusing styles in other components such as inline alerts.

Reply:
I'd like to continue discussing this conversation once we've decided on an approach to the `getFromReadable` conversation below.

Reply:
As discussed over private chat, this can be taken up in a subsequent PR